### DISCUSSION TITLE
How to create a rectangular gradient shadow?

### DISCUSSION THREAD

**Discussion by LaniusSoftware** [SixLabors/ImageSharp.Drawing#342] [Q&A] [answered]
I'm looking to create a shadow for a rectangle (either inner or outer) but I can't achieve this with the built-in gradients.
To give you an example, this is an inner shadow:
![image](https://github.com/user-attachments/assets/aed081d1-02e6-4faf-9628-edb99c9c799f)
And this is an outer shadow:
![image](https://github.com/user-attachments/assets/a8970bfe-993b-423c-8859-eea1c813de33)
I tried reverse engineering a LinearGradientBrush but it's not very well documented and GradientBrushApplicator is an internal class which I can't access.
Is there some way to achieve this using this library?

**Comment by JimBobSquarePants** [comment, marked as answer]
No brush is going to allow what you want since the angle of the gradient would have to change as you draw along each face of your rectangle. I would draw a rectangle; blur it then clip out what you don't want.

**Reply by LaniusSoftware** [reply]
Thank you, works perfectly.
I'd like to ask you if the code I wrote to handle this case could be improved or not. Maybe I'm missing some way to do this using less lines of code.
```
var canvas = new Image<Rgba32>((int)shadedRect.Width + (2 * canvasMargin), (int)shadedRect.Height + (2 * canvasMargin));
var shadowRect = new RectangleF(canvasMargin, canvasMargin, shadedRect.Width, shadedRect.Height);
canvas.Mutate(context =>
{
    var options = new DrawingOptions();
    options.GraphicsOptions.AlphaCompositionMode = PixelAlphaCompositionMode.Clear;
    context.Draw(Color.Black, 10f, shadowRect);
    context.BoxBlur();
    if (isOuterShadow)
    {
        context.Fill(options, Color.Transparent, shadowRect);
    }
    else
    {
        var topClip = new RectangleF(0f, 0f, canvas.Width, shadowRect.Y);
        context.Fill(options, Color.Transparent, topClip);
        var rightClip = new RectangleF(shadowRect.X + shadowRect.Width, 0f, canvasMargin, canvas.Height);
        context.Fill(options, Color.Transparent, rightClip);
        var bottomClip = new RectangleF(0f, shadowRect.Y + shadowRect.Height, canvas.Width, canvasMargin);
        context.Fill(options, Color.Transparent, bottomClip);
        var leftClip = new RectangleF(0f, 0f, shadowRect.X, canvas.Height);
        context.Fill(options, Color.Transparent, leftClip);
    }
});
```
If not, it can serve as a solution to anyone looking for such functionality.
